x/gov/client/cli/tx.go

```diff
@@ -135,6 +136,38 @@ Where proposal.json contains:
 	return cmd
 }
 
+// NewCmdCancelProposal implements submitting a cancel proposal transaction command.
+func NewCmdCancelProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "cancel-proposal [proposal-id]",
+		Short:   "Cancel governance proposal before the voting period ends.",
+		Args:    cobra.ExactArgs(1),
+		Example: fmt.Sprintf(`$ %s tx gov cancel-proposal 1 --from mykey`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			// validate that the proposal id is a uint
```

Comment:
should we also validate that the proposal id corresponds to an existing proposal id so cancellation doesn't fail after the cancel_proposal passes?

Reply:
We are checking for the proposal  while canceling [here](https://github.com/cosmos/cosmos-sdk/pull/13010/files#diff-e25a55efe77bf0d11d86d054c74f7fd44201572f4abb72a79c9dbd4e374ea25dR100)